File: Android/source-code/AppStore/app/src/main/java/org/buildmlearn/appstore/models/Apps.java
package org.buildmlearn.appstore.models;

import android.graphics.Bitmap;
import android.os.Parcel;
import android.os.Parcelable;

/**
 * Model class of the app, currently used for store apps, rendered from the Store XML.
 */
public class Apps implements Parcelable {
    public String Name,Description,Category,Author,AuthorEmail,AppIcon,Type;
    public Bitmap BAppIcon;
    public String[] Screenshots;

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(Name);
        dest.writeString(Description);
        dest.writeString(Category);
        dest.writeString(Type);
        dest.writeString(Author);
        dest.writeString(AuthorEmail);
        dest.writeString(AppIcon);
        dest.writeInt(Screenshots.length);
        for (String Screenshot : Screenshots) dest.writeString(Screenshot);

    }
    /**
     * This is used to regenerate your object. All Parcelables must have a CREATOR that implements these two methods
     */
    public static final Parcelable.Creator<Apps> CREATOR = new Parcelable.Creator<Apps>() {
        public Apps createFromParcel(Parcel in) {
            return new Apps(in);
        }

        public Apps[] newArray(int size) {
            return new Apps[size];
        }
    };

    /**
     * Example constructor that takes a Parcel and gives you an object populated with it's values
     * @param in Parcelable object
     */
    private Apps(Parcel in) {
        Name=in.readString();
        Description=in.readString();
        Category=in.readString();
        Type=in.readString();
        Author=in.readString();
        AuthorEmail=in.readString();
        AppIcon = in.readString();
        int l=in.readInt();
        Screenshots=new String[l];
        for(int i=0;i<l;i++)Screenshots[i]=in.readString();
    }
    public Apps()
    {
        Name="";Description="";Category="";Type="";Author="";AuthorEmail="";AppIcon="";
        Screenshots=null;
    }
}

File: WP/source-code/AppStore/AppStore/Models/Apps.cs
namespace AppStore.Models
{
    /// <summary>
    /// It is model for the Info App-Template.
    /// </summary>
    class Apps
    {
        public string Name { get; set; }
        public string Author { get; set; }
        public string AuthorEmail { get; set; }
        public string Description { get; set; }
        public string AppIcon { get; set; }
        public string Category { get; set; }
        public string []Screenshots { get; set; }
        public string Type { get; set; }

        /// <summary>
        /// Public Constructor
        /// </summary>
        /// <param name="name">Name of the app</param>
        /// <param name="description">Description of the app</param>
        /// <param name="author">Author of the app</param>
        /// <param name="authoremail">AuthorEmail </param>
        /// <param name="appicon">AppIcon</param>
        /// <param name="screenshots">Screenshots</param>
        /// <param name="type">Type of the app</param>
        /// <param name="category">Category of the app</param>
        public Apps(string name,string description, string author, string authoremail,string appicon,string []screenshots,string type,string category)
        {
            Name = name.Trim();
            Description = description.Trim();
            Author = author.Trim();
            AuthorEmail = authoremail.Trim();
            AppIcon = appicon.Trim();
            Screenshots = screenshots;
            Category = category.Trim();
            Type = type.Trim();
        }

    }
}
